File: 3731-find-missing-elements/3731-find-missing-elements.cpp
class Solution {
public:
    vector<int> findMissingElements(vector<int>& nums) {
        vector<int> res;
        bool present[101] = {};
        int minVal = INT_MAX, maxVal = INT_MIN;

        for (int ele : nums) {
            minVal = min(minVal, ele);
            maxVal = max(maxVal, ele);
            present[ele] = true;
        }

        for (int i = minVal; i <= maxVal; i++) {
            if (!present[i]) res.push_back(i);
        }
        return res;
    }
};

File: 3731-find-missing-elements/3731-find-missing-elements.py
class Solution:
    def findMissingElements(self, nums: List[int]) -> List[int]:
        present = [False] * 101
        minVal, maxVal = float('inf'), float('-inf')

        for ele in nums:
            minVal = min(minVal, ele)
            maxVal = max(maxVal, ele)
            present[ele] = True

        return [i for i in range(minVal, maxVal + 1) if not present[i]]